ergotree-interpreter/src/sigma_protocol/private_input.rs

```diff
@@ -76,13 +78,57 @@ impl From<Scalar> for DlogProverInput {
     }
 }
 
+/// Diffie-Hellman tuple and secret
+/// Used in a proof that of equality of discrete logarithms (i.e., a proof of a Diffie-Hellman tuple):
+/// given group elements g, h, u, v, the proof convinces a verifier that the prover knows `w` such
+/// that `u = g^w` and `v = h^w`, without revealing `w`
+#[derive(PartialEq, Debug, Clone)]
+pub struct DhTupleProverInput {
+    /// Diffie-Hellman tuple's secret
+    pub w: Scalar,
+    /// Diffie-Hellman tuple
+    pub common_input: ProveDhTuple,
+}
+
+impl DhTupleProverInput {
+    /// Create random secret and Diffie-Hellman tuple
+    #[allow(clippy::many_single_char_names)]
+    pub fn random() -> Self {
```

Comment:
Is it used in tests only? If yes, better to specify it in the comments (or move the function into tests?)

Reply:
Yes. I moved it to tests. 
There is also `DlogProverInput::random()` which is used in tests but 
it's public and exposed in WASM layer(JS/TS). Do you prefer to remove `DlogProverInput::random()` from all public APIs as well?

Reply:
ah then it is reasonable to have both random() public and exposed. Please check that randomness is of cryptographic quality.

Reply:
I switched to `ThreadRng` PRNG (more cryptographically secure than `OsRng`) and restored public `DhTupleProverInput::random()`
[`ThreadRng`](https://rust-random.github.io/rand/rand/rngs/struct.ThreadRng.html) is [`StgRng`](https://rust-random.github.io/rand/rand/rngs/struct.StdRng.html) (ChaCha block cipher with 12 as of July 2021) seeded from [`OsRng`](https://rust-random.github.io/rand/rand/rngs/struct.OsRng.html) which is a random number generator that retrieves randomness from the operating system.